lib/types.js

```diff
@@ -382,7 +390,9 @@ module.exports = function (expect) {
                 }
                 return numberOfSimilarKeys >= requiredSimilarKeys;
             });
-        }
+        },
+        uniqueKeys: utils.uniqueStringsAndSymbols,
+        valueForKey: null
```

Comment:
Could you provide a default implementation for `valueForKey` that just goes `obj[key]` instead of checking for its existence above?

I'm not totally sure a `valueForKey` hook would be the ideal hook for an iterable type (where random access might be a problem), but it's clearly better than what we have now. Maybe we should prefix it with an underscore to signal that it's subject to change, though.

Ideally we should also change the `array-like` type to use it.